src/utils/breadcrumb-utils.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { Badge, BreadcrumbItem } from '@patternfly/react-core';
+import styles from '@patternfly/react-styles/css/components/Breadcrumb/breadcrumb';
+import { ApplicationSwitcher } from '../components/ApplicationDetails/ApplicationSwitcher';
+import { WorkspaceSwitcher } from '../components/ApplicationDetails/WorkspaceSwitcher';
+import { useWorkspaceInfo } from './workspace-context-utils';
+
+export const useApplicationBreadcrumbs = (appDisplayName = null, withLink = true) => {
+  const params = useParams();
+  const applicationName = params.appName;
+
+  const { workspace } = useWorkspaceInfo();
+
+  return [
+    <Badge key="badge" isRead>
+      WS
+    </Badge>,
+    <span key="badge-divider" className={styles.breadcrumbItemDivider} />,
+    <BreadcrumbItem key="workspace-link" to="#">
+      <Link className="pf-c-breadcrumb__link" to="#">
+        {workspace}
+      </Link>
+    </BreadcrumbItem>,
+    <WorkspaceSwitcher key="workspace" />,
+    <span key="workspace-divider" className={styles.breadcrumbItemDivider}>
+      |
+    </span>,
+    <BreadcrumbItem key="app-link">
+      <Link
```

Comment:
Nit: to avoid react warnings
```suggestion
    <BreadcrumbItem key="workspace-link" to="#" component="div">
      <Link className="pf-c-breadcrumb__link" to="#">
        {workspace}
      </Link>
    </BreadcrumbItem>,
    <WorkspaceSwitcher key="workspace" />,
    <span key="workspace-divider" className={styles.breadcrumbItemDivider}>
      |
    </span>,
    <BreadcrumbItem key="app-link" component="div">
      <Link
```